journal.py

```diff
@@ -181,45 +183,27 @@ def edit_entry(request):
     if request.authenticated_userid:
         if request.method == 'POST':
             try:
-                update_entry(request)
-            except psycopg2.Error:
+                Entry.update(request)
+            except SQLAlchemyError:
                 return HTTPInternalServerError()
-            cursor = request.db.cursor()
-            cursor.execute(READ_ENTRY, (request.params.get('id', None), ))
-            keys = ('id', 'title', 'text', 'created')
-            entry = dict(zip(keys, cursor.fetchone()))
-            entry['text'] = markdown.markdown(entry['text'],
-                                              extensions=['codehilite', 'fenced_code'])
-            entry['created'] = entry['created'].strftime('%b. %d, %Y')
-            return entry
+            entry = Entry.by_id(id=request.params.get('id', None))
+            json_entry = {
+                'id': entry.id,
+                'title': entry.title,
+                'text': markdown.markdown(entry.text,
+                                          extensions=['codehilite', 'fenced_code']),
+                'created': entry.created.strftime('%b. %d, %Y')
+            }
+            return json_entry
```

Comment:
Here's another place where a method on the Entry object would allow you to reproduce this same dict without needing to re-write the code in a dozen places.